client/me/help/help-contact-form/index.jsx

```diff
@@ -93,21 +93,22 @@ export const HelpContactForm = React.createClass( {
 		this.props.valueLink.requestChange( this.state );
 	},
 
-	getSiteSlug() {
-		if ( this.props.selectedSiteSlug ) {
-			return this.props.selectedSiteSlug;
+	getSiteId() {
+		if ( this.props.selectedSiteId ) {
+			return this.props.selectedSiteId;
 		}
 
 		const primarySite = sites.getPrimary();
 		if ( primarySite ) {
-			return primarySite.slug;
+			return primarySite.ID;
 		}
 
 		return null;
 	},
 
 	setSite( siteSlug ) {
-		this.setState( { siteSlug } );
+		const site = sites.getSite( siteSlug );
```

Comment:
For anyone reading we only store the selected site in Redux at the moment. Avoiding a sites list usage will require us to make an additional requestSite() call. If we go forward with the changes here, we'll want to clean this up later.